webrender/src/tiling.rs

```diff
@@ -861,34 +866,35 @@ impl AlphaBatchKey {
             flags: AlphaBatchKeyFlags::new(TransformedRectKind::AxisAligned,
                                            false),
             blend_mode: BlendMode::Alpha,
-            color_texture_id: TextureId::invalid(),
-            mask_texture_id: TextureId::invalid(),
+            textures: BatchTextures::no_texture(),
         }
     }
 
     fn primitive(kind: AlphaBatchKind,
                  flags: AlphaBatchKeyFlags,
                  blend_mode: BlendMode,
-                 color_texture_id: TextureId,
-                 mask_texture_id: TextureId)
+                 textures: BatchTextures)
                  -> AlphaBatchKey {
         AlphaBatchKey {
             kind: kind,
             flags: flags,
             blend_mode: blend_mode,
-            color_texture_id: color_texture_id,
-            mask_texture_id: mask_texture_id,
+            textures: textures,
         }
     }
 
     fn is_compatible_with(&self, other: &AlphaBatchKey) -> bool {
         self.kind == other.kind &&
             self.flags == other.flags &&
             self.blend_mode == other.blend_mode &&
-        (self.color_texture_id == TextureId::invalid() || other.color_texture_id == TextureId::invalid() ||
-             self.color_texture_id == other.color_texture_id) &&
-            (self.mask_texture_id == TextureId::invalid() || other.mask_texture_id == TextureId::invalid() ||
-             self.mask_texture_id == other.mask_texture_id)
+            (!self.textures.colors[0].is_valid() || !other.textures.colors[0].is_valid() ||
```

Comment:
Can we add a (inline) function to check if two textures are compatible? That would make this function a lot easier to read.